dex/elrond_dex_farm/src/rewards.rs

```diff
@@ -44,36 +40,113 @@ pub trait RewardsModule {
         to_mint
     }
 
-    fn calculate_reward_for_given_liquidity(
-        &self,
-        liquidity: &Self::BigUint,
-        enter_amount: &Self::BigUint,
-        farm_token_supply: &Self::BigUint,
-        virtual_reserves: &Self::BigUint,
-        actual_reserves: &Self::BigUint,
-    ) -> Self::BigUint {
-        let big_zero = Self::BigUint::zero();
-        let reward_amount = self.calculate_reward_amount_current_block();
-        let total_reserves = virtual_reserves + actual_reserves + reward_amount;
+    fn generate_aggregated_rewards(&self, reward_token_id: &TokenIdentifier) {
+        let reward_minted = self.mint_per_block_rewards(&reward_token_id);
+        let fees = self.reset_temporary_fee_storage();
+        let total_reward = reward_minted + fees;
 
-        let worth = if farm_token_supply > &0 {
-            liquidity * &total_reserves / farm_token_supply.clone()
-        } else {
-            big_zero.clone()
-        };
+        if total_reward > 0 {
+            self.increase_reward_reserve(&total_reward);
+            self.update_reward_per_share(&total_reward);
+        }
+    }
+
+    fn reset_temporary_fee_storage(&self) -> Self::BigUint {
+        let current_block = self.blockchain().get_block_nonce();
```

Comment:
is this still needed ? these are fees from penalties ?

Reply:
these are the fees from swap pairs. those should be distributed with the same logic if we decide to send fees from swap pairs to farm contracts